cmd/kubeadm/app/cmd/init.go

```diff
@@ -242,28 +244,40 @@ func (i *Init) Run(out io.Writer) error {
 		return fmt.Errorf("couldn't parse kubernetes version %q: %v", i.cfg.KubernetesVersion, err)
 	}
 
+	// Get directories to write files to; can be faked if we're dry-running
```

Comment:
I'm still kind of flummoxed why we are passing args around vs. having a centralized configuration object.  For example, for testing and validation I may want to dump the state of config during a dryRun for *, which would include the tempdirs b/c I want to inspect the certs.